yt/chyt/controller/internal/api/api.go

```diff
@@ -840,3 +840,22 @@ func (a *API) SetSecrets(ctx context.Context, alias string, secrets map[string]a
 	)
 	return err
 }
+
+func (a *API) Resume(ctx context.Context, alias string) error {
+	if err := a.CheckExistence(ctx, alias, true /*shouldExist*/); err != nil {
+		return err
+	}
+	if err := a.CheckPermissionToOp(ctx, alias, yt.PermissionManage); err != nil {
+		return err
+	} // TODO: maybe yt.PermissionUse?
+
+	oplet, err := a.getOpletFromCypress(ctx, alias, nil, a.cfg.AgentInfo)
+	if err != nil {
+		return err
+	}
+
+	if err = oplet.ResumeOp(ctx, "api call"); err != nil {
+		return err
+	}
+	return oplet.Pass(ctx)
```

Comment:
I don't like resuming the op and making the pass here.

We do so for `untracked` operations, because there is no controller who tracks them.

But here the API and the controller both read/write the persistent state and it may lead to conflicts (you may loose part of persistent state).

The `resume` operation is not easily expressible via speclet, but as a workaround I suggest to introduce a separate option like `latest_resume_time` - and just set it to `time.Now()` here. The actual `resume` logic should be moved to the controller based on this value. Another alternative is to introduce the option `resume_stamp` (not sure about the naming) and set it to a random int64 value. Then you can check if the value has been changed in the controller and resume operation.

Moreover, the `resume` command is jupyt-related, let's move it to `jupyt` folder. And actually we'are implementing a `restart` command now. It looks like `restart` can be an alternative to `resume` operation. Do we really need the `resume` command?

Reply:
Well, I thought about this option... Can't say I like it (nor the other options), but you are right, it fits the current strawberry logic better. 

> the resume command is jupyt-related, let's move it to jupyt folder

It's a command for all controller families that support scaling to 0, isn't it? I don't think jupyt will be unique for that matter.